### PR TITLE
Adding DCO file and updating contributing.md with details

### PR DESCRIPTION
**Pull request by leemills83**
## Hey, I just made a Pull Request!

Adding DCO file and DCO information to CONTRIBUTING.md as PR ahead of DCO going live on 03/01/21.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4513-fyyn98ggkmxc4jxnbtsmttlfgtz6y9ja.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60269c2afcb22db6f55441ed

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -80,6 +80,22 @@ Also be sure to skim through our [ADRs](https://github.com/backstage/backstage/t
 
 If there are any updates in `markdown` file please make sure to run `yarn run lint:docs`. Though it is checked on `lint-staged`. It is required to install [vale](https://docs.errata.ai/vale/install) separately and make sure it is accessed by global command.
 
+## DCO
```
Comment (adamdmharvey) [inline CONTRIBUTING.md:83]: Scanning headers, I may not have known what this was - new users may not be as familiar. Could this be spelled out in the header, and then in the body (where you have already) you can list it and note it's abbreviation parenthetically first time before then using it further?

**Comment by Fox32** [comment]
I already played around a bit with it in my latest PRs. Maybe some tips: 

**Forgot to add it to the latest commit?** [Amend it](https://stackoverflow.com/questions/13043357/git-sign-off-previous-commits/15667644#15667644):
```shell
git commit --amend --signoff
```

**Forgot it on multiple commits in your branch?** [Rebase the last n commits and add the signoff](https://stackoverflow.com/questions/13043357/git-sign-off-previous-commits/63884006#63884006) (here last two commits):

```shell
git rebase --signoff HEAD~2
```

If you have already pushed you branch to a remote, you might have to force push: `git push -f`.

**Comment by OrkoHunter** [comment]
@Fox32 Thanks! I expect us writing this comment on a lot of PRs starting next month. Perhaps the DCO bot should suggest these to a new contributor, IMO.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Rugvip) [inline CONTRIBUTING.md:83]:
```suggestion
## Developer Certificate of Origin
```

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4513-fyyn98ggkmxc4jxnbtsmttlfgtz6y9ja.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60269c2afcb22db6f55441ed

**Review by jhaals — APPROVED** [review]

**Review by Rugvip — APPROVED** [review]

**Pull request merged**